public/js/monitor.js

```diff
@@ -1,90 +1,122 @@
-    "featuredBreach": "User scanned email from featured breach page.",
-    "foundBreaches": "User scanned additional email.",
-    "noBreaches": "User scanned email address after scanning email with no associated breaches.",
-    "landingPage": "User submitted email from landing page.",
-
+    "eventAction": "",
+  },
+  "ShowAdditional": {
+    "eventCategory": "Show Additional Breaches",
+    "eventAction": "Button Click",
   },
   "Pageview": {
     "eventCategory":"Views",
-    "eventAction": "Viewed",
-    "featuredBreach": "Featured Breach",
-    "foundBreaches": "List of compromised accounts.",
-    "noBreaches": "No associated breaches.",
-    "landingPage": "Default landing page without featured breach.",
-    "confirmAccount": "Your account is confirmed page.",
+    "eventAction": "",
+  },
+  "Download" : {
+    "eventCategory": "Download",
+    "eventAction": "Firefox Download",
+  },
+  "Link": {
+    "eventCategory": "Link Clicks",
+    "eventAction": "Link Click",
   },
 };
 
+// determines which page the event occurs and is sent to google analytics in ga_sendPing() as the eventLabel.
+
 function ga_getLocation() {
-  if (document.getElementById("featured-breach-landing")) {
-      return("featuredBreach");
+  if (document.querySelector(".landing-content")) {
+    if (document.getElementById("featured-breach-landing")) {
+      return "Featured Breach Page";
+    }
+    return "Landing Page";
   }
   if (document.getElementById("found-breaches")) {
-      return("foundBreaches");
+      return "Scan Results - found breaches.";
   }
   if (document.getElementById("no-breaches")) {
-      return("noBreaches");
+      return "Scan Results - no breaches.";
+  }
+  if (document.getElementById("unsubscribe-form")) {
+    return "Unsubscribe Page";
   }
   if (document.getElementById("unsubscribe-survey-form")) {
-    return("unsubscribePage");
+    return "Unsubscribe Survey";
+  }
+  if (document.getElementById("confirmation")) {
+    return "Account Confirmation Page";
+  }
+  if (document.getElementById("error-message")) {
+    return "Error Page";
   }
-    return("landingPage");
 }
 
-function ga_sendPing(string, eventLabel = false) {
-  if(typeof(ga) !== "undefined") {
 
-    const event = events[string];
+function ga_sendPing(eventDescription, eventLabel) {
+  if(typeof(ga) !== "undefined") {
+    const event = eventList[eventDescription];
     const eventCategory = event["eventCategory"];
     const eventAction = event["eventAction"];
 
-    if (!eventLabel) {
-      const eventLocation = ga_getLocation();
-      eventLabel = event[eventLocation];
-      ga("send", "event", eventCategory, eventAction, eventLabel);
-      return;
+    if(eventLabel) {
+      if(eventDescription.includes("UnsubscribeSurvey")) {
+        return ga("send", "event", eventCategory, eventAction, eventLabel);
+      }
+      return ga("send", "event", eventCategory, eventAction, eventLabel, ga_getLocation());
```

Comment:
Per the [event tracking implementation docs](https://developers.google.com/analytics/devguides/collection/analyticsjs/events#implementation), I'm afraid this will try to set the location as the `eventValue`?

Reply:
nice catch! You're right- 'value' has to be a number. Will send location as dimension.